### PR TITLE
Allow to edit all the assets

### PR DESCRIPTION
**Pull request by macano**

### CODE REVIEW

File: webtool/gradle/common/src/commonMain/kotlin/Pilot.kt [hunk 1]
```diff
@@ -46,10 +46,20 @@ data class Pilot(
         else -> throw Exception("Unknown type")
     }
 
-    fun replaceHouseHoldGroup(index: Int, householdGroup: HouseholdGroup): Pilot {
-        val newGroups = this.householdGroups.toMutableList()
-        newGroups[index] = householdGroup
-        return copy(householdGroups = newGroups)
+    // Generalized replace function
```
Comment (Erikvv) [inline webtool/gradle/common/src/commonMain/kotlin/Pilot.kt:49]: Nit: I don't understand the appeal of this. Why not have 

```kotlin
fun replaceBattery(battery: Battery, index: Int): Pilot 
    = copy(batteries = batteries.replaceAt(index, battery))
```

etc

**Review by Erikvv — APPROVED** [review]
Tested and works. Good for merge.

File: webtool/frontend/src/components/assets/battery.tsx [hunk 2]
```diff
@@ -64,11 +66,11 @@ export const BatteryForm: FunctionComponent<{
...
                     <label className="form-label" htmlFor="capacity_kWh">Capaciteit (kWh)</label>
-                    <input className="form-input" type="number" id="capacity_kWh" name="capacity_kWh" defaultValue={100} />
+                    <input className="form-input" type="number" id="capacity_kWh" name="capacity_kWh" defaultValue={ initialData?.capacity_kWh || 100} />
                 </div>
                 <div className="radix-grid">
                     <label className="form-label" htmlFor="peakPower_kW">Vermogen (kW)</label>
-                    <input className="form-input" type="number" id="peakPower_kW" name="peakPower_kW" defaultValue={100} />
+                    <input className="form-input" type="number" id="peakPower_kW" name="peakPower_kW" defaultValue={ initialData?.peakPower_kW || 100} />
                 </div>
                 <CostSection hideCostPerKwh={true} />
```
[the remark this answers is not in the document]
Comment (macano) [reply webtool/frontend/src/components/assets/battery.tsx:75]: missed it. Done

[the remark this answers is not in the document]
Comment (Erikvv) [reply webtool/frontend/src/components/assets/battery.tsx:75]: Nothing is pushed yet

[the remark this answers is not in the document]
Comment (Erikvv) [reply webtool/frontend/src/components/assets/battery.tsx]: Is now pushed and works but forgot solar panels

File: webtool/frontend/src/components/assets/battery.tsx [hunk 3]
```diff
@@ -77,3 +79,37 @@ export const BatteryForm: FunctionComponent<{
         </Card>
     )
 }
+
+export const BatteriesDisplayEdit: FunctionComponent<{
+    pilot: Pilot,
+    onChange: (pilot: Pilot) => void,
+}> = ({pilot, onChange}) => {
+    const [selected, setSelected] = useState<Battery | null>(null);
```
[the remark this answers is not in the document]
Comment (Erikvv) [reply webtool/frontend/src/components/assets/battery.tsx:87]: Was discussed.

File: webtool/frontend/src/components/assets/battery.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (macano) [reply webtool/frontend/src/components/assets/battery.tsx]: Added to Solar Panels

**Pull request merged**